lib/core/string.toit

```diff
@@ -1391,6 +1403,39 @@ class StringSlice_ extends string:
   compute-hash_ -> int:
     #primitive.core.blob-hash-code
 
+class StringByteSlice_ implements io.Data:
+  str_ / String_
+  from_ / int
+  to_   / int
+
+  constructor .str_ .from_ .to_:
+
+  // TODO(florian): this method is only here for backwards-compatability.
+  // Some methods used to take 'any' and then take the 'size' of it.
+  // Once we have migrated all these locations to use 'io.Data' and 'byte-size', it
+  // can be removed.
+  size -> int:
+    return byte-size
+
+  byte-size -> int:
+    return to_ - from_
+
+  byte-slice from/int to/int -> io.Data:
+    actual-from := from_ + from
+    actual-to := from_ + to
+    if not from_ <= actual-from <= actual-to <= to_: throw "OUT_OF_BOUNDS"
+    return StringByteSlice_ str_ actual-from actual-to
+
+  byte-at index/int -> int:
+    actual-index := from_ + index
```

Comment:
Missing bounds check.

Reply:
There will be a bounds check in the string, but
done.

Reply:
That bounds check in the string won't work for from_ = 5, index = -1